File: sentimental-mario-less/mario.py
def get_int(prompt):
    """Prompt the user for a positive integer between 1 and 8, inclusive.
    If the input is not a positive integer between 1 and 8, re-prompt the user.
    """
    while True:
        try:
            value = int(input(prompt))
            if 1 <= value <= 8:
                return value
            else:
                print("Please enter a positive integer between 1 and 8, inclusive.")
        except ValueError:
            print("Please enter a positive integer between 1 and 8, inclusive.")


# Prompt the user for the height of the half-pyramid
height = get_int("Height: ")

# Print the half-pyramid
for i in range(height):
    # Print the spaces
    for j in range(height - i - 1):
        print(" ", end="")
    # Print the hashes
    for j in range(i + 1):
        print("#", end="")
    # Print a newline
    print()

File: mario-less/mario.c
#include <cs50.h>  // Include the cs50 header file
#include <stdio.h>  // Include the standard input/output header file

int main(void)
{
    // Prompt the user for the height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Build the pyramid
    for (int i = 1; i <= height; i++)
    {
        // Print the leading spaces
        for (int j = 1; j <= height - i; j++)
        {
            printf(" ");
        }

        // Print the hashes
        for (int j = 1; j <= i; j++)
        {
            printf("#");
        }

        // Print a newline character
        printf("\n");
    }

    // Return 0 to indicate that the program ran successfully
    return 0;
}